### DISCUSSION TITLE
Assign shaders to USD's in Maya....

### DISCUSSION THREAD

**Discussion by ragtag** [Autodesk/maya-usd#2786] [General]
I've been trying to find a way to assign Arnold shaders to individual objects within a USD in Maya, and the options there seem somewhat limited. I've tried three approaches:

1. When creating a Stage and bringing in the USD's, it seems like it's only possible to assign a shader to the whole thing, and not individual objects within that USD. You can, of course, convert it to Maya data assign/edit shaders, and then back to USD, but I was kind of hoping to avoid that kind of workflow.
2. If you use Bifrost to bring in USDs, you can actually override and assign shaders, but as far as I can tell, this only works with things like usd_preview_surface and not aiStandtardSurface.
3. Using aiStandIn's and operators does work, but then you're not really getting any of the benefits of USD and might as well use Alembic or .ass files.

Is there a better way to do this?

The end goal is to be able to run modeling/rigging/animation and shading/texturing in parallel, and then combine cached animation with the latest shaders at the lighting stage.

**Comment by wallworm** [comment]
Sorry we took so long to respond to this question. This is a very high priority for us and we are actively exploring ways to make lookdev and shading in USD as easy and natural as working with Maya data.
